### PR TITLE
Fixed a bug when oc_process executes fail_json.

### PR DESCRIPTION
**Pull request by kwoodson**

### CODE REVIEW

**Comment by ashcrow** [comment]
aos-ci-test

**Comment by ashcrow** [comment]
[test]

File: roles/lib_openshift/src/class/oc_process.py [hunk 1]
```diff
@@ -175,11 +175,11 @@ def run_ansible(params, check_mode):
                 update = True
 
         if not update:
-            return {"changed": update, "results": api_rval, "state": "present"}
+            return {"changed": update, "results": api_rval, "state": state}
 
         for cmd in rval:
             if cmd['returncode'] != 0:
-                return {"failed": True, "changed": update, "results": rval, "state": "present"}
+                return {"failed": True, "changed": update, "msg": rval, "state": state}
```
Comment (tbielawa) [inline roles/lib_openshift/src/class/oc_process.py]: I see the subject is
> Fixed a bug when oc_process executes fail_json

and the commit message says
> Fixed a bug. Ansible requires a msg param when module.fail_json.

But the code changes only change the value of the `state` return parameter from strings to what the module `state` parameter was set to when the module is invoked.

Looks like the `msg` param is only added in one return statement.

So the other changes are just regular bug fixes (return the actual requested state, not a hard-coded string) and the change in `src/class/oc_process.py` on line ~182 adds the actual `msg` param.

If that's all correct and intended then +1.

**Comment by tbielawa** [comment]
Seeing this in the "[Indication 1](https://ci.openshift.redhat.com/jenkins/job/test_pull_request_openshift_ansible_extended_conformance_install/99/consoleFull#-72929089656cbb9a5e4b02b88ae8c2f77)" error logs


```
• Failure [122.222 seconds]
[builds][Conformance] oc new-app
/go/src/github.com/openshift/origin/test/extended/builds/new_app.go:52
  should succeed with a --name of 58 characters [It]
  /go/src/github.com/openshift/origin/test/extended/builds/new_app.go:44

  Expected error:
      <*errors.StatusError | 0xc42099ab00>: {
          ErrStatus: {
              TypeMeta: {Kind: "", APIVersion: ""},
              ListMeta: {SelfLink: "", ResourceVersion: ""},
              Status: "Failure",
              Message: "the server has asked for the client to provide credentials (get builds)",
              Reason: "Unauthorized",
              Details: {
                  Name: "",
                  Group: "",
                  Kind: "builds",
                  Causes: [
                      {
                          Type: "UnexpectedServerResponse",
                          Message: "Unauthorized",
                          Field: "",
                      },
                  ],
                  RetryAfterSeconds: 0,
              },
              Code: 401,
          },
      }
      the server has asked for the client to provide credentials (get builds)
  not to have occurred

  /go/src/github.com/openshift/origin/test/extended/builds/new_app.go:39
```

In the "[Indication 2](https://ci.openshift.redhat.com/jenkins/job/test_pull_request_openshift_ansible_extended_conformance_install/99/consoleFull#-8865011056cbb9a5e4b02b88ae8c2f77)" error logs

```
Summarizing 2 Failures:

[Fail] [builds][Conformance] oc new-app [It] should succeed with a --name of 58 characters 
/go/src/github.com/openshift/origin/test/extended/builds/new_app.go:39

[Fail] deploymentconfigs viewing rollout history [Conformance] [It] should print the rollout history 
/go/src/github.com/openshift/origin/test/extended/deployments/deployments.go:454

Ran 190 of 725 Specs in 1426.749 seconds
FAIL! -- 188 Passed | 2 Failed | 0 Pending | 535 Skipped 

Ginkgo ran 1 suite in 23m48.468373404s
Test Suite Failed
```

[Indication 3 logs](https://ci.openshift.redhat.com/jenkins/job/test_pull_request_openshift_ansible_extended_conformance_install/99/consoleFull#82220402358b6e51eb7608a5981914356) 

```
SUCCESS! -- 22 Passed | 0 Failed | 0 Pending | 703 Skipped Apr  3 11:43:01.539: INFO: Error running cluster/log-dump.sh: fork/exec /data/src/github.com/openshift/origin/vendor/k8s.io/kubernetes/cluster/log-dump.sh: no such file or directory
PASS

Ginkgo ran 1 suite in 12m0.780970874s
Test Suite Passed
[INFO] Dumping container logs to /tmp/openshift/conformance/logs/containers
[INFO] Exiting
[ERROR] PID 25977: test/extended/conformance.sh:51: `[[ -n "${LOG_DIR:-}" && -d "${LOG_DIR-}" ]]` exited with status 1.
[INFO] 		Stack Trace: 
[INFO] 		  1: test/extended/conformance.sh:51: `[[ -n "${LOG_DIR:-}" && -d "${LOG_DIR-}" ]]`
[INFO]   Exiting with code 1.
make: *** [test-extended] Error 1
++ export status=FAILURE
++ status=FAILURE
+ set +o xtrace
########## FINISHED STAGE: FAILURE: RUN EXTENDED TESTS ##########
```

**Comment by ashcrow** [comment]
@tbielawa I'm not sure if that's related or not. I think saw it with another PR earlier

**Comment by kwoodson** [comment]
@ashcrow @tbielawa, welcome to the flakes boys!
flake https://github.com/openshift/origin/issues/13067

**Comment by sdodson** [comment]
[test]

**Comment by abutcher** [comment]
Supersedes #3836

**Comment by ashcrow** [comment]
[merge]

**Comment by kwoodson** [comment]
[merge]

**Pull request merged**